### ISSUE TITLE
Formatting of negative numbers

### ISSUE THREAD

**Issue by barry-pawssum** [scurker/currency.js#95]
When formatting a negative number with formatWithSymbol:true, the minus sign is placed after the currency symbol, i.e. "$-10.00". I think it's more commonly seen placed before the currency symbol, i.e. "-$10.00". Can this be changed, or at least an option added to choose where the minus sign should be placed?

**Comment by scurker** [comment]
This is definitely an issue, but I want to think about this some more.

Generally, negative formats can vary depending on the field. General accounting in the US typically formats negative values with parens `($1.00)` while other fields will have the sign before the symbol `-$1.00`, after the value `€1,00-`, or even between the symbol in the value `€-1,00`. Currently, currency.js doesn't handle formatting in a very elegant way so I'd like a better approach to solve general formatting issues.

**Comment by knodel12** [comment]
Is there any update to this?  I have a scenario at my work where I need to format negative numbers with parenthesis for accounting purposes and negative signs for non-accounting related issues.  This library is nearly perfect for our use case, but this is a major issue for us.

**Comment by agrohs** [comment]
Same issue here, any update by chance?

**Comment by scurker** [comment]
Sorry for any delay folks. This is coming soon, but I'm working on a few internal things before release. Changes are currently available on the `next` tag if you want to have an early look.

**Comment by scurker** [comment]
This is now available on the latest release, `v1.2.0`.

`pattern` and `negative` pattern are now available as options to customize the formatting output using `!` and `#` as replacement values for the symbol and amount respectively.

Examples Usage:

```javascript
var customCurrency = value => currency(value, { pattern: '! #', negativePattern: '(! #)', formatWithSymbol: true });

customCurrency(1234567.89).format();  // => "$ 1,234,567.89"
customCurrency(-1234567.89).format(); // => "($ 1,234,567.89)"
```
